app/middleware/searchValidator.js

```diff
@@ -0,0 +1,33 @@
+const log = require('../lib/logger');
+const renderer = require('../middleware/renderer');
+const searchValidator = require('../lib/searchValidator');
+
+function setSearchLabel(res, search) {
+  if (search) {
+    // eslint-disable-next-line no-param-reassign
+    res.locals.searchLabel = 'Enter your GP surgery name';
+  }
+}
+
+function validateSearch(req, res, next) {
+  const search = res.locals.search;
+
+  log.info('validate-search-start');
+  const validationResult = searchValidator(search);
+  log.info('validate-search-end');
+
+  // eslint-disable-next-line no-param-reassign
+  res.locals.search = validationResult.input;
+
+  if (validationResult.errorMessage) {
+    log.info({ search }, 'Search failed validation');
+    // eslint-disable-next-line no-param-reassign
+    res.locals.errorMessage = validationResult.errorMessage;
+    setSearchLabel(res, search);
+    renderer.findHelp(req, res);
```

Comment:
'the current empty search returns a 500 error, that is something I will be looking next' - you spotted the error, so will fix that now and add a test